cadc-vos-server-alt/src/main/java/org/opencadc/vospace/server/async/RecursiveDeleteNodeRunner.java

```diff
@@ -0,0 +1,358 @@
+            throw new RuntimeException("BUG: NodePersistence implementation not found with JNDI key " + jndiNodePersistence, oops);
+        }
+    }
+
+    @Override
+    public void setJobUpdater(JobUpdater jobUpdater) {
+        this.jobUpdater = jobUpdater;
+    }
+
+    @Override
+    public void setJob(Job job) {
+        this.job = job;
+    }
+
+    @Override
+    public void setSyncOutput(SyncOutput syncOutput) {
+        // not used
+    }
+
+    @Override
+    public void run() {
+        log.debug("RUN RecursiveDeleteNodeRunner");
+        logInfo = new JobLogInfo(job);
+
+        String startMessage = logInfo.start();
+        log.info(startMessage);
+
+        long t1 = System.currentTimeMillis();
+        doit();
+        long t2 = System.currentTimeMillis();
+
+        logInfo.setElapsedTime(t2 - t1);
+
+        String endMessage = logInfo.end();
+        log.info(endMessage);
+    }
+
+    private void doit() {
+        try {
+            // set the phase to executing
+            ExecutionPhase ep = jobUpdater.setPhase(
+                job.getID(), ExecutionPhase.QUEUED, ExecutionPhase.EXECUTING, new Date());
+
+            if (ep == null) {
+                throw new IllegalStateException(
+                        "Could not change the job phase from " + ExecutionPhase.QUEUED
+                                + " to " + ExecutionPhase.EXECUTING);
+            }
+
+            VOSURI nodeURI = null;
+            for (Parameter param : job.getParameterList()) {
+                if (param.getName().equalsIgnoreCase("nodeURI")) {
+                    nodeURI = new VOSURI(URI.create(param.getValue()));
+                    break;
+                }
+            }
+
+            if (nodeURI == null) {
+                throw new IllegalArgumentException("nodeURI argument required");
+            }
+
+            log.debug("node: " + nodeURI);
+
+            PathResolver pathResolver = new PathResolver(nodePersistence, vospaceAuthorizer, true);
+            String nodePath = nodeURI.getPath();
+            Node serverNode = pathResolver.getNode(nodePath);
+            if (serverNode == null) {
+                throw NodeFault.NodeNotFound.getStatus(nodePath);
+            }
+
+            ContainerNode parent = serverNode.parent;
+            Subject caller = AuthenticationUtil.getCurrentSubject();
+            if (!vospaceAuthorizer.hasSingleNodeWritePermission(parent, caller)) {
+                throw NodeFault.PermissionDenied.getStatus(nodePath);
+            }
+
+            if (serverNode instanceof ContainerNode) {
+                try {
+                    deleteContainer((ContainerNode) serverNode, caller);
+                } catch (JobAbortedException ex) {
+                    // nothing to do here
+                }
+            } else {
+                try {
+                    nodePersistence.delete(serverNode);
+                    log.debug("Recursively deleted data node " + nodePath);
```

Comment:
this debug message will be misleading because this is the non-recursive option

Reply:
Removed. The idea was to show that the file was deleted as a result of the recursive delete but that it's already illustrated by the name of the file.